src/components/[guild]/EditGuild/components/Admins/Admins.tsx

```diff
@@ -66,7 +66,7 @@ const Admins = () => {
     const ownerOption = {
       ...(options.find((o) => o.value === ownerAddress) ?? {
         value: ownerAddress,
-        label: shortenHex(ownerAddress),
+        label: ownerAddress.length === 42 ? shortenHex(ownerAddress) : ownerAddress,
```

Comment:
There's an `ADDRESS_REGEX` regex which we use to validate addresses in our application, you could use that here too